### PR TITLE
Extracted generic fedora client code.

### PR DESCRIPTION
**Pull request by mikedurbin**
https://jira.duraspace.org/browse/FCREPO-1815

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -52,6 +52,7 @@
     <spring.version>4.1.4.RELEASE</spring.version>
     <fcrepo.version>4.4.0</fcrepo.version>
     <fcrepo-transform.version>4.4.0</fcrepo-transform.version>
+    <fcrepo-java-client.version>4.4.1-SNAPSHOT</fcrepo-java-client.version>
```
Comment (acoburn) [inline pom.xml]: I'd really prefer to depend on a non-SNAPSHOT version of the java client.

Comment (mikedurbin) [inline pom.xml]: I thought about that... 

We're going to get this into jenkins so the snapshot releases exist in the maven repo, then I guess I should roll a non-snapshot release, then update this PR?  

Alternately, we could leave this as snapshot, but then when a release of fcrepo-camel is made, you'll be warned about snapshot dependencies by the release tooling and at that time, lock it down to a non-snapshot version.  We're certainly going to face times in the future when the snapshot version of fcrepo-camel *MUST* point to a snapshot version of fcrepo-java-client, or we'll have to roll releases every time we need to make a change in that client code for use by dependent applications.

Comment (acoburn) [inline pom.xml]: +1 to the first option. The problem with the second option is that releases _must_ then be coordinated across projects, and it was decided (for good reasons) to allow independent projects to be versioned and released independently.

Comment (mohideen) [inline pom.xml]: Should this be 0.2.0-SNAPSHOT instead of 4.4.1-SNAPSHOT?
https://github.com/fcrepo4-exts/fcrepo-java-client/blob/master/pom.xml#L16

Comment (acoburn) [inline pom.xml]: @mohideen yes it should

Comment (acoburn) [inline pom.xml]: @mohideen see: https://jira.duraspace.org/browse/FCREPO-1858 and the two corresponding PRs which address this.

Comment (mohideen) [inline pom.xml]: @acoburn ++

**Comment by acoburn** [comment]
This was merged earlier via https://github.com/fcrepo4-exts/fcrepo-camel/commit/1695a1d51d3552ac4599c9c5f90b3bb1526db5be

**Pull request closed without merging**